electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -189,6 +198,32 @@ def merge_exchanges(
 
         return exchanges
 
+    @staticmethod
+    def update_exchanges(
+        exchanges: "ExchangeList", new_exchanges: "ExchangeList", logger: Logger
+    ) -> "ExchangeList":
+        """Given a new batch of exchanges, update the existing ones."""
+        if len(new_exchanges) == 0:
+            return exchanges
+        elif len(exchanges) == 0:
+            return new_exchanges
+
+        for new_event in new_exchanges.events:
+            if new_event.datetime in exchanges:
```

Comment:
Hum, I'm thinking that the datetime indexing is not super explicit, since this is initially a list of events.
This should probably be explicited in the class description.

Reply:
That is what I do in the follow up PR, but I can do it directly in here as well and then update the other PR to just contain the logic changes I want to make.

Reply:
In my follow up PR I am changing this to be a dict internally to streamline a lot of the logic without changing how it works externally.

It drastically simplifies the logic of all the magic functions.

That PR is still a draft though as it hasn't beef finalized or fully tested yet.

Reply:
Added a it to the top level EventList class as there is where the magic functions are implemented now. Let me know if you want to make it even more clear.